PWGHF/Tasks/taskMcValidation.cxx

```diff
@@ -363,6 +392,10 @@ struct HfTaskMcValidationGen {
               !RecoDecay::isMatchedMCGen(mcParticles, particle, PDGArrayParticle[iD], std::array{+kOmegaMinus, +kPiPlus}, true)) {
             continue;
           }
+          if (iD == B0ToDminusPi &&
+              !RecoDecay::isMatchedMCGen(mcParticles, particle, PDGArrayParticle[iD], std::array{-o2::constants::physics::Pdg::kDPlus, +kPiPlus}, true)) {
```

Comment:
```suggestion
              !RecoDecay::isMatchedMCGen<true, false>(mcParticles, particle, PDGArrayParticle[iD], std::array{-o2::constants::physics::Pdg::kDPlus, +kPiPlus}, true)) {
```
Here we need to account for the B0/B0bar oscillations. This should also be reflected in the check done before on the 4-prong final state.